ios/RNSScreenStack.mm

```diff
@@ -978,10 +979,23 @@ - (void)mountChildComponentView:(UIView<RCTComponentViewProtocol> *)childCompone
       @(index),
       @([childComponentView.superview tag]));
 
-  [_reactSubviews insertObject:(RNSScreenView *)childComponentView atIndex:index];
-  ((RNSScreenView *)childComponentView).reactSuperview = self;
+  RNSScreenView *screenChildComponent = (RNSScreenView *)childComponentView;
+  [_reactSubviews insertObject:screenChildComponent atIndex:index];
+
+  // Since the view may be recycled we don't want to remount snapshot to the hierarchy.
+  // Thus, we want to reset view to initial view and force `setPushViewControllers` to be called.
+  if ([NSStringFromClass([screenChildComponent.controller.view class]) isEqualToString:@"_UIReplicantView"]) {
```

Comment:
This check can ve tricky if the name of private class for snapshot changes. Wouldn't it be better to check if it is not `RNSScreenView`?